README.md

```diff
@@ -114,6 +114,77 @@ The element also emits **events** as a user interacts with it. This is how you c
 |`change`       |`string`      |Dispatched with every keystroke as the user types (not debounced).|
 |`error`        |`Error`       |Dispatched if an error occures during the request (for example if the rate limit was exceeded.)|
 
+
+
+## Customization
+
+We did our best to design the element in such a way that it can be used as is in a lot of different contexts without needing to adjust how it looks. However, there certainly can be situations where customization is necessary. The element supports three different customization APIs:
+
+1. Custom CSS (variables)
+2. A string template as well as
+3. A row template
+
+### 1. Custom CSS
+
+We use CSS variables for almost all properties that you would want to customize. This includes the font family, background or shadow of the input and the hover state for a result just to name a few. For a list of all available variables please check the source CSS file directly ([`/src/autocomplete/autocomplete.css`](src/autocomplete/autocomplete.css)).
+
+You can adjust these variables by placing a `<style>` tag _inside_ the element, like so:
+
+```html
+<ge-autocomplete api_key="…">
+  <style>
+     :host {
+       --input-bg: salmon;
+       --input-color: green;
+       --loading-color: hotpink;
+    }
+  </style>
+</ge-autocomplete>
+```
+
+**Important:** While it is technically possible to override the actual classnames the element uses internally, we do not consider those part of the public API. That means they can change without a new major version, which could break your customization. The CSS variables on the other hand are specifically meant to be customized, which is why they will stay consistent even if the internal markup changes.
+
+If you would like to customize a property for which there is no variable we’d be happy to accept a pull-request or issue about it.
+
+### 2. String Template
+
+If you want to customize how a feature is turned into a string for rendering (in the results as well as the input field after it was selected), you can define a custom string template. This allows you to use the [lodash template language][_template] to access every property of the feature to build a custom string.
+
+```html
+<ge-autocomplete api_key="…">
+  <template string>
+    ${feature.properties.name} (${feature.properties.id}, ${feature.properties.source})
+  </template>
+</ge-autocomplete>
+```
+
+**Important:** Make sure to return a plain string here, no HTML. The reason for this is that this template will also be used in the input field itself after a result has been selected, which doesn’t support HTML.
```

Comment:
We will _probably_ want to support HTML in the dropdown menu at some point.

In addition to just custom styling in general, highlighting which part of the results match the input string can be an extremely helpful UI device. It's kinda hard to see with the all-black styling, but our [leaflet demo](https://geocodeearth.github.io/autocomplete-demo/) supports this out of the box, for example.

It's 100% fine to not support HTML for now, but maybe we don't want to lean too heavily in the other direction in case of future functionality?